functions/src/definitions/userHandlers.js

```diff
@@ -569,6 +594,21 @@ async function onTextListReceipt(messageObj, platform = "whatsapp") {
           )
           await sleep(3000)
           break
+
+        case "referral":
+          const code = (await db.collection("users").doc(from).get()).get(
+            "referralId"
+          )
+          if (code) {
+            response = responses.REFERRAL.replace(
+              "{{link}}",
+              `https://ref.checkmate.sg/${code}`
+            )
+          } else {
+            functions.logger.error(`Referral code not found for ${from}`)
```

Comment:
`response` should be assigned something in this situation. Not sure how `await sendWhatsappTextMessage` will behave if response is `undefined`

Reply:
Fixed